### ISSUE TITLE
webpack --watch does not run loaders when a semicolon is added/removed

### ISSUE THREAD

**Issue by gajus** [webpack/webpack#1272]
Subject:

``` js
let foo

console.log('abc', foo);
```

I am using:
- webpack
- eslint-loader

I am running webpack using `webpack --watch --progress`.

The following output is generated:

```
ERROR in ./foo.js

/foo.js
  1:7  error  Missing semicolon  semi

✖ 1 problem (1 error, 0 warnings)
```

So far so good.

I edit the subject file:

``` js
let foo;

console.log('abc', foo);
```

`webpack` CLI `--progress` indicates that package is re-built. **However, error output remains.**

I edit the subject file (edit the string):

``` js
let foo;

console.log('def', foo);
```

`webpack` CLI `--progress` indicates that package is re-built. The error disappears.

Therefore, it appears that adding/removing a semicolon does not trigger loaders.

**Comment by dnutels** [comment]
Same here, with nearly the same setup, only with babel-loader. There is no reaction on any change webpack deems insignificant. removal of `;` (even if that would've failed eslint, removal or addition of spaces.

More significant changes, like adding or removing a `var a = 10;` statement, do trigger re-loading.

For reference, here is the `config` file:

```
module.exports = {
    entry: {
        app: PATH.resolve(root, 'src/bundle.js'),
    },
    output: {
        path: PATH.resolve(root, 'public/js'),
        filename: 'bundle.js'
    },
    module: {
        loaders: [
            {
                test: /\.js$/,
                exclude: /(node_modules|bower_components)/,
                loader: 'babel'
            },
            {
                test: /\.js$/,
                exclude: /(node_modules|bower_components)/,
                loader: 'eslint-loader'
            }
        ]        
    }
};
```

**Comment by nmschulte** [comment]
I believe I am experiencing this issue as well (the issue that webpack doesn't respect file timestamps, or even simple content-hashing, when detecting changes).  At first I thought I was just changing the file too quickly after the previous change was detected (and that may actually be an issue too, it's hard to tell), but the commentors above pointed me in the right direction: I'm usually modifying superfluous whitespace, as our linter (eslint) is setup to error on these issues and that will cause webpack to fail.

What's the change detection algorithm/heuristic, and why doesn't it seem to be anything sane?

**Comment by gajus** [comment]
I have not figured out whats the problem with `webpack.NewWatchingPlugin` (the default watching plugin), but you can force webpack to use the old file watching logic if you add `new webpack.OldWatchingPlugin()` plugin. The `OldWatchingPlugin` does not suffer from the same issue.

**Comment by gajus** [comment]
https://github.com/webpack/watchpack/issues/16

**Comment by bluenote10** [comment]
Would it be possible to re-open this issue?

I'm still facing this issue on webpack 2.3.3 / Ubuntu, and the corresponding issue (#16 on watchpack)[https://github.com/webpack/watchpack/issues/16] has been closed as well, pointing out that it is a webpack issue.

In can confirm that it is not an inotify issue (`/proc/sys/fs/inotify/max_user_watches` set to 524288), because I can see a very short progress indication. However, webpack seems to decide not to recompile the file for some reason.

**Comment by rafalp** [comment]
Apparently eslint will not run against your files if change has no effect on output AST. Perhaps a way to extempt loader from this check could be added, like `always: true`?

I've also ran into this issue few days back as I was experimenting with the Webpack 2 for first time, and then I've resorted to running separate watching process. Still, its just a workaround untill this gets resolved in better way.

**Comment by engelen** [comment]
+1

**Comment by nmschulte** [comment]
_bump._

**Comment by rafalp** [comment]
@nmschulte note that this issue was fixed in Webpack 3.5.4:

> Warnings and errors contribute to hash, which shows stats on warning-only change

**Comment by ianmays** [comment]
still seeing this issue on ^3.10.0
